core/src/kura.rs

```diff
@@ -962,4 +990,209 @@ mod tests {
+
+            kura.store_block(block_genesis.clone());
+            kura.store_block(block);
+            // Add wait so that previous block is written to the block store
+            thread::sleep(Duration::from_secs(1));
+            kura.replace_top_block(block_soft_fork.clone());
+            kura.store_block(block_next.clone());
+        }
+
+        // Reinitialize kura and check that correct blocks are loaded
+        {
+            let (kura, block_count) = Kura::new(&Config {
+                init_mode: InitMode::Strict,
+                store_dir: iroha_config::base::WithOrigin::inline(
+                    temp_dir.path().to_str().unwrap().into(),
+                ),
+                debug_output_new_blocks: false,
+            })
+            .unwrap();
+
+            assert_eq!(block_count.0, 3);
+
+            assert_eq!(
+                kura.get_block_by_height(nonzero!(1_usize)),
+                Some(Arc::new(block_genesis.into()))
+            );
+            assert_eq!(
+                kura.get_block_by_height(nonzero!(2_usize)),
+                Some(Arc::new(block_soft_fork.into()))
+            );
+            assert_eq!(
+                kura.get_block_by_height(nonzero!(3_usize)),
+                Some(Arc::new(block_next.into()))
+            );
+        }
+    }
+
+    #[allow(clippy::too_many_lines)]
+    fn create_blocks(rt: &tokio::runtime::Runtime) -> Vec<CommittedBlock> {
+        let mut blocks = Vec::new();
+
+        let (leader_public_key, leader_private_key) = KeyPair::random().into_parts();
+        let peer_id = PeerId::new("127.0.0.1:8080".parse().unwrap(), leader_public_key);
+        let topology = Topology::new(vec![peer_id]);
+
+        let chain_id = ChainId::from("00000000-0000-0000-0000-000000000000");
+
+        let (genesis_id, genesis_key_pair) = gen_account_in("genesis");
+        let genesis_domain_id = DomainId::from_str("genesis").expect("Valid");
+        let genesis_domain = Domain::new(genesis_domain_id).build(&genesis_id);
+        let genesis_account = Account::new(genesis_id.clone()).build(&genesis_id);
+        let (account_id, account_keypair) = gen_account_in("wonderland");
+        let domain_id = DomainId::from_str("wonderland").expect("Valid");
+        let domain = Domain::new(domain_id).build(&genesis_id);
+        let account = Account::new(account_id.clone()).build(&genesis_id);
+
+        let live_query_store = LiveQueryStore::test();
+        let live_query_store = {
+            let _rt_guard = rt.enter();
+            live_query_store.start()
+        };
+        let state = State::new(
+            World::with(
+                [domain, genesis_domain],
+                [account, genesis_account],
+                [],
+                UniqueVec::new(),
+            ),
+            Kura::blank_kura_for_testing(),
+            live_query_store,
+        );
+
+        let executor = {
+            let wasm_blob = iroha_wasm_builder::Builder::new("../default_executor")
```

Comment:
this seems to heavy for a unit test

Reply:
i haven't find a way around it

Reply:
Because in order for genesis block to be valid it have to have valid `Upgrade` instruction.

Reply:
Would it be possible to write a mock executor with wasm text representation in the same way it's done in core unit tests? https://github.com/hyperledger/iroha/blob/58c943e44eea09e6f0c80c487547d0ff11f9eccb/core/src/smartcontracts/wasm.rs#L1776

Or maybe use the same executor blob that is used in integration tests' genesis (`defaults/executor.wasm`)?